Mage.Sets/src/mage/cards/w/WoodElemental.java

```diff
@@ -0,0 +1,133 @@
+
+    public WoodElemental(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId,setInfo,new CardType[]{CardType.CREATURE},"{3}{G}");
+        this.subtype.add(SubType.ELEMENTAL);
+        this.power = new MageInt(0);
+        this.toughness = new MageInt(0);
+
+        // As Wood Elemental enters the battlefield, sacrifice any number of untapped Forests.
+        this.addAbility(new AsEntersBattlefieldAbility(new WoodElementalEffect()));
+
+        // Wood Elemental's power and toughness are each equal to the number of Forests sacrificed as it entered the battlefield.
+                this.addAbility(new SimpleStaticAbility(Zone.ALL, new InfoEffect("{this}'s power and toughness are each equal to the number of Forests sacrificed as it entered the battlefield")));
+    }
+
+    public WoodElemental(final WoodElemental card) {
+        super(card);
+    }
+
+    @Override
+    public WoodElemental copy() {
+        return new WoodElemental(this);
+    }
+}
+
+class WoodElementalEffect extends OneShotEffect {
+    
+    private static final FilterControlledPermanent filter = new FilterControlledPermanent("untapped Forests you control");
+    
+    static {
+        filter.add(Predicates.not(new TappedPredicate()));
+        filter.add(new SubtypePredicate(SubType.FOREST));
+    }
+
+    public WoodElementalEffect() {
+        super(Outcome.Sacrifice);
+        staticText = "sacrifice any number of untapped Forests";
+    }
+
+    public WoodElementalEffect(final WoodElementalEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public WoodElementalEffect copy() {
+        return new WoodElementalEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Card sourceCard = game.getCard(source.getSourceId());
+        if (controller != null && sourceCard != null) {
+            Target target = new TargetControlledPermanent(0, Integer.MAX_VALUE, filter, true);
+            if (target.canChoose(source.getSourceId(), source.getControllerId(), game) 
+                    && controller.chooseTarget(Outcome.Detriment, target, source, game)) {
+                if (!target.getTargets().isEmpty()) {
+                    int sacrificedForests = target.getTargets().size();
+                    game.informPlayers(controller.getLogName() + " sacrifices " + sacrificedForests + " untapped Forests for " + sourceCard.getLogName());
+                    for (UUID targetId : target.getTargets()) {
+                        Permanent targetPermanent = game.getPermanent(targetId);
+                        if (targetPermanent != null) {
+                            targetPermanent.sacrifice(source.getSourceId(), game);
+                        }
+                    }
+                    game.addEffect(new SetPowerToughnessSourceEffect(sacrificedForests, sacrificedForests, Duration.WhileOnBattlefield, SubLayer.SetPT_7b), source);
```

Comment:
I would use Duration.CUSTOM, so the effect should be discarded if Wood Elemental has left the battlefeild.
With WhileOnBattlefield you get multiple continuous effects active at the same time when Wood Elemental has left and returned to the battlefield. P/T result probably would be correct.